enhancements/kubelet/oci-image-volume.md

```diff
@@ -0,0 +1,180 @@
+  - mrunalp
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: 2025-05-07
+last-updated: 2025-05-07
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/OCPNODE-2929
+  - https://issues.redhat.com/browse/OCPNODE-2575
+see-also:
+  - N/A
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+# OCI Image Volume
+
+## Summary
+
+In Kubernetes 1.33, support for OCI Image Volume went to Beta.
+This enhancement is to enable this feature in OpenShift.
+See [KEP-4639: Summary](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4639-oci-volume-source#summary)
+
+## Motivation
+
+See [KEP-4639: Motivation](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4639-oci-volume-source#motivation)
+
+### User Stories
+
+See [KEP-4639: User Stories](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4639-oci-volume-source#user-stories-optional)
+
+### Goals
+
+- Enable ImageVolume feature gate for OpenShift.
+
+### Non-Goals
+
+- This proposal does not aim to replace existing VolumeSource types.
+- This proposal does not address other use cases for OCI objects beyond directory sharing among containers in a pod.
+- Mounting thousands of images in a single pod.
+
+## Proposal
+
+1. Enable ImageVolume feature gate in 4.20
+
+    - This feature gate is disabled by default in Kubernetes 1.33, so we need to explicitly enable it for OCP
+
+2. Separate Graduation Approach between OCI Images and OCI Artifacts.
+
+    - **OCI Image Mount**: Promote to GA
+    - **OCI Artifact Mount**: Begin as DevPreview
+        - OCI Artifact functionality is less mature and requires further refinement
+        - Add a new CRI-O option to toggle OCI artifact mount support
+        - Disable OCI artifact mount by default
+
+### Workflow Description
+
+#### OCI Image
+
+Since OCI image mount will be GA in 4.20. No feature gate configuration is needed by users. Creating a pod with volumes.image will pull the image and mount in the container.
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+spec:
+  containers:
+  - name: test
+    image: registry.k8s.io/e2e-test-images/echoserver:2.3
+    volumeMounts:
+    - name: volume
+      mountPath: /volume
+  volumes:
+  - name: volume
+    image:
+      reference: quay.io/crio/artifact:v1
+      pullPolicy: IfNotPresent
+```
+
+#### OCI Artifact
+
+While OCI artifact is in dev preview, it fails to mount if a user specifies OCI artifact in reference. To enable the mount, users need to add this drop-in CRI-O configuration.
```

Comment:
I'm not sure if we should call it `dev preview`, which may imply that we change the OpenShift API. We can just state that it has to be manually enabled once proven to be stable in CRI-O.

Reply:
similar discussion https://redhat-internal.slack.com/archives/GK6BJJ1J5/p1745584306680429 .
@haircommander wdyt?

Reply:
Okay, so dev preview seems fine to my understanding now.

Reply:
yeah I typically use Dev Preview as a shorthand for not in the API yet